tensorflow/core/grappler/clusters/BUILD

```diff
@@ -22,7 +22,8 @@ config_setting(
 tf_cuda_library(
     name = "utils",
     srcs = ["utils.cc"],
-    hdrs = ["utils.h"],
+    hdrs = ["utils.h",
+            "//tensorflow/core/common_runtime:device_factory.h",],
```

Comment:
Add `"//tensorflow/core/common_runtime:device_factory"` to deps  instead.

In most cases, we only add headers to hdrs if they correspond to this specific target.

Reply:
we tried to make it as deps, however, some ut will fail because device_factory has global variable(device_factories) which will be compiled into several libraries. that will make device_factories not consistent.
```
std::unordered_map<string, FactoryItem>& device_factories() {
  static std::unordered_map<string, FactoryItem>* factories =
      new std::unordered_map<string, FactoryItem>;
  return *factories;
}
```

Reply:
The header shouldn't be needed now since device_factory.h now a part of `"//tensorflow/core:framework",`. If it doesn't work without any new deps, try adding `"//tensorflow/core:framework_internal",`.

Reply:
I have remove it

Reply:
I had to rollback the change that moved device_factory to framework due to a test failure. However, now I fixed and resubmitted. Thank you for removing the dep!